eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -224,127 +230,81 @@ object ChannelCodecs extends Logging {
       ("localNextHtlcId" | uint64overflow) ::
       ("remoteNextHtlcId" | uint64overflow) ::
       ("originChannels" | originsMapCodec) ::
-      ("remoteNextCommitInfo" | either(bool, waitingForRevocationCodec, publicKey)) ::
+      ("remoteNextCommitInfo" | either(boolean, waitingForRevocationCodec, publicKey)) ::
       ("commitInput" | inputInfoCodec) ::
-      ("remotePerCommitmentSecrets" | ShaChain.shaChainCodec) ::
+      ("remotePerCommitmentSecrets" | byteAligned(ShaChain.shaChainCodec)) ::
```

Comment:
The `shaChainCodec` isn't byte aligned, and there may be room for optimization there. cc @sstone

Reply:
I think we should make it byte-aligned, but the question is whether to do it "outside" (what you just did), or inside the shachain codec (use `bool8` instead of `bool`)?

Reply:
Apparently we can be more optimal, the codec should naturally be byte-aligned. Since this is sensitive code I'd rather leave that optimisation to another PR.